ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -76,15 +87,26 @@ targetPeers PeerSelectionActions{readPeerSelectionTargets}
               targets,
               ledgerStateJudgement,
               bootstrapPeersFlag,
-              hasOnlyBootstrapPeers
-            } =
+              hasOnlyBootstrapPeers,
+              useGenesisFlag
+              } =
     Guarded Nothing $ do
-      targets' <- readPeerSelectionTargets
+      configurationTargets <- readPeerSelectionTargets
+      -- following in case the configuration is reloaded while condition is true,
+      -- so that big ledger peer targets don't get wiped out accidentally (or by malice)
+      -- or for the first time when we observe that the chain is style in Genesis mode
+      let configurationAmend = if ledgerStateJudgement == TooOld && useGenesisFlag
+                               then configurationTargets { targetNumberOfActivePeers               = 0, -- to keep unneeded connections to a minimum
+                                                           targetNumberOfKnownBigLedgerPeers       = governor_GENESIS_KNOWN_BIG_LEDGER_PEERS,
+                                                           targetNumberOfEstablishedBigLedgerPeers = governor_GENESIS_ESTABLISHED_BIG_LEDGER_PEERS,
+                                                           targetNumberOfActiveBigLedgerPeers      = governor_GENESIS_ACTIVE_BIG_LEDGER_PEERS }
+                               else configurationTargets
+                                
       check ( isNodeAbleToMakeProgress bootstrapPeersFlag
                                        ledgerStateJudgement
                                        hasOnlyBootstrapPeers
-            && targets' /= targets
-            && sanePeerSelectionTargets targets'
+            && configurationAmend /= targets
```

Comment:
I imagined a scenario where the targets are reloaded manually by a user/SPO while we are in Genesis mode and on a stale chain. They could zero out active big ledger peers, which would circumvent Genesis guarantees. This code checks for that and overrides those targets. It seemed sensible to me, because if the user wants this behavior then it is not Genesis mode and so the node should be restarted with that flag set to false.

Reply:
Yeah I guess it is okay, since the user might have exactly the same config